packages/amplify_flutter/lib/src/categories/amplify_auth_category.dart

```diff
@@ -247,7 +247,7 @@ class AuthCategory {
   ///
   /// Accepts plugin-specific, advanced [options] for the request
   Future<UpdateUserAttributeResult> updateUserAttribute({
-    required String userAttributeKey,
+    required Object userAttributeKey,
```

Comment:
I think `Object` is sort of a confusing public API. If the goal is to avoid putting a cognito-specific class (CognitoUserAttribute) in the category interface, I think we could make a generic UserAttribute class and then either create a cognito-specific implementation of that class, or add the cognito specific pieces via an extension.

Example of how `CognitoUserAttribute` could be split into a generic `UserAttribute` and cognito-specific extension

```dart
// generic class for the category interface
class UserAttribute {
  const UserAttribute._(this.key, {this.readOnly = false});

  /// The JSON key for this attribute.
  final String key;

  /// Whether this attribute is read only
  final bool readOnly;

  /// Creates a custom attribute.
  const UserAttribute.custom(String key)
      : key = 'custom:$key',
        readOnly = false;

  // other immutable methods ...
}

// cognito implementation
extension CognitoUserAttribute on UserAttribute {
  static const address = UserAttribute._('address');

  // other attributes ...

  /// Parses the given Cognito attribute key.
  static parse(String key) {
    return values.firstWhere(
      (attr) => attr.key == key,
      orElse: () => UserAttribute._(key),
    );
  }

  static const values = [
    address,
    // other attributes ...
  ];
}
```